### PR TITLE
Add metamod functionality to ribametamod

### PR DESCRIPTION
**Pull request by HendrikKok**

### CODE REVIEW

**Comment by HendrikKok** [comment]
This branch contains all work for implement the MetaMod functionality in the RibaMetaMod-driver.

This branch contains:
1- Add all MetaSWAP interactions 
2- Refactor on mapping: all mapping functionality is moved to a separate class for a better overview.
2- Refactor exchanges to equal form

File: imod_coupler/drivers/ribametamod/ribametamod.py [hunk 1]
```diff
@@ -12,31 +12,34 @@
...
 
     base_config: BaseConfig  # the parsed information from the configuration file
-    ribamod_config: RibaMetaModConfig  # the parsed information from the configuration file specific to Ribametamod
+    ribametamod_config: RibaMetaModConfig  # the parsed information from the configuration file specific to Ribametamod
     coupling: Coupling  # the coupling information
 
     timing: bool  # true, when timing is enabled
     mf6: Mf6Wrapper  # the MODFLOW 6 kernel
     ribasim: RibasimApi  # the Ribasim kernel
+    msw: MswWrapper  # the MetaSWAP XMI kernel
```
Comment (Hofer-Julian) [inline imod_coupler/drivers/ribametamod/ribametamod.py:35]:
```suggestion
    msw: MswWrapper  # the MetaSWAP kernel
```

File: imod_coupler/drivers/ribametamod/ribametamod.py [hunk 2]
```diff
@@ -126,75 +145,176 @@ def couple(self) -> None:
...
 
-        # One time step in MODFLOW 6
-        self.mf6.update()
+        # exchange drainage fluxes from MODFLOW 6 to Ribasim
+        self.exchange_rib2mod()
 
+        # Update Ribasim until current time of MODFLOW 6
+        self.ribasim.update_until(self.get_current_time() * days_to_seconds(self.delt))
+
+    def update_MODFLOW6_MetaSWAP(self) -> None:
```
Comment (Hofer-Julian) [inline imod_coupler/drivers/ribametamod/ribametamod.py:227]:
```suggestion
    def update_modflow6_metaswap(self) -> None:
```

Let's try to stick to snake case for method names: https://developer.mozilla.org/en-US/docs/Glossary/Snake_case

File: imod_coupler/drivers/ribametamod/ribametamod.py [hunk 3]
```diff
@@ -126,75 +145,176 @@ def couple(self) -> None:
...
+            if has_converged:
+                logger.debug(f"MF6-MSW converged in {kiter} iterations")
+                break
+        self.mf6.finalize_solve(1)
+
+        # finish MODFLOW 6 - MetaSWAP timestep
+        self.mf6.finalize_time_step()
+        self.msw.finalize_time_step()
+
+    def do_MODFLOW6_MetaSWAP_iter(self, sol_id: int) -> bool:
```
Comment (Hofer-Julian) [inline imod_coupler/drivers/ribametamod/ribametamod.py:248]:
```suggestion
    def do_modflow6_metaswap_iter(self, sol_id: int) -> bool:
```

**Review by Hofer-Julian — APPROVED** [review]
I had a few minor comments, but the changes look good to me, in general

**Pull request merged**